pylearn2/space/__init__.py

```diff
@@ -548,6 +548,129 @@ def _format_as_impl(self, is_numeric, batch, target_space):
         raise NotImplementedError("%s does not implement _format_as_impl()." %
                                   type(self))
 
+    def undo_np_format_as(self, batch, space):
+        """
+        Returns a numeric batch (e.g. a numpy.ndarray or scipy.sparse sparse
+        array), with formatting from space undone.
+
+        This is just a wrapper around self._undo_format_as, with an extra check
+        to throw an exception if <batch> is symbolic. Formatting to space
+
+        Parameters
+        ----------
+        batch : numpy.ndarray, or one of the scipy.sparse matrices.
+            Array which lies in this space.
+        space : Space
+            Space to undo formatting from.
+
+        Returns
+        -------
+        WRITEME
+            The formatted batch
+        """
+
+        self._check_is_numeric(batch)
+
+        return self._undo_format_as(is_numeric=True,
+                                    batch=batch,
+                                    space=space)
+
+    def undo_format_as(self, batch, space):
+        """
+        Returns a symbolic batch (e.g. a theano.tensor or theano.sparse
+        array), with formatting from space undone.
+
+        This is just a wrapper around self._undo_format_as, with an extra check
+        to throw an exception if <batch> is symbolic. Formatting to space
+
+        Parameters
+        ----------
+        batch : numpy.ndarray, or one of the scipy.sparse matrices.
+            Array which lies in this space.
+        space : Space
+            Space to undo formatting from.
+
+        Returns
+        -------
+        WRITEME
+            The formatted batch
+        """
+        self._check_is_symbolic(batch)
+
+        return self._undo_format_as(is_numeric=False,
+                                    batch=batch,
+                                    space=space)
```

Comment:
Same thing here, what is done by _undo_format_as(is_numeric=False, ...) could simply be put here.